[CI] Breakup h200 tests

LucasWilkinson opened the pull request:
The "Distributed Tests (H200)" takes a very long time to run; and has been flaky due to environmental OOMs. Break it up so retries are less costly and move some tests to cheaper H100s (open to suggestions here)

We should really trim-down or optimize the `tests/distributed/test_sequence_parallel.py` and `tests/compile/distributed/test_fusions_e2e.py` tests in future PRs but this is a start.

## Overall Test Times

| Test | Duration | Result |
|------|----------|--------|
| `tests/compile/distributed/test_async_tp.py` | 10:58 (658s) | 28 passed |
| `tests/compile/distributed/test_sequence_parallelism.py` | 15:51 (951s) | 48 passed |
| `tests/compile/distributed/test_fusion_all_reduce.py` | 1:27 (87s) | 8 passed |
| `tests/compile/distributed/test_fusions_e2e.py` (not Llama-4) | 1:00:47 (3648s) | 44 passed, 2 skipped, 20 deselected |
| `tests/distributed/test_sequence_parallel.py` | 37:14 (2235s) | 40 passed |
| `tests/distributed/test_context_parallel.py` | 9:46 (586s) | 10 passed |
| `examples/offline_inference/data_parallel.py` | ~33s | completed |
| `tests/v1/distributed/test_dbo.py` | 2:24 (144s) | 2 passed |

PR:
<img width="1621" height="71" alt="image" src="https://github.com/user-attachments/assets/3a8d2773-0db5-4919-a69c-169a0bbbe649" />

NOTE: fast follow to accelerate these here: https://github.com/vllm-project/vllm/pull/30513

---

> [!NOTE]
> Reduces flakiness and runtime by reorganizing CI and refactoring fusion tests.
> 
> - CI: add optional `Hopper Fusion E2E Tests (H100)` (single-GPU and 2xH100) and `Sequence Parallel Tests (H100)`; trim `Distributed Tests (H200)` to a smaller set; adjust filters/commands for fusion suites in `.buildkite/test-pipeline.yaml`.
> - Testing refactor: extract shared helpers into `tests/compile/fusion_test_utils.py`; simplify `tests/compile/distributed/test_fusions_e2e.py` to use utilities and focus on TP2 cases; move/expand attention fusion E2E checks into `tests/compile/test_fusion_attn.py`; add RMSNorm+group quant fusion E2E in `tests/compile/test_fusion.py`.
> - Update reference in `vllm/env_override.py` comment to point to `tests/compile/test_fusion_attn.py::test_attn_quant`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5aefa70761e4bda4ad0471465c33cca5acc31a92. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> <sup>[Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) is generating a summary for commit d500caea6c477d94a04eb6e86d76b86da2ad5a94. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Reduces CI flakiness/runtime and streamlines fusion testing.
> 
> - CI: add optional `Hopper Fusion E2E Tests (H100)` (single- and 2xH100) and `Sequence Parallel Tests (H100)`; trim `Distributed Tests (H200)` to a smaller set; tweak commands/filters for fusion suites in `.buildkite/test-pipeline.yaml`.
> - Testing refactor: introduce `tests/compile/fusion_test_utils.py` with shared models/helpers; simplify `tests/compile/distributed/test_fusions_e2e.py` to TP2-focused checks (attn+quant, allreduce+RMSNorm, async TP) using the utilities; expand E2E coverage in `tests/compile/test_fusion_attn.py` (add attn quant E2E) and `tests/compile/test_fusion.py` (add RMSNorm+group FP8 quant E2E); add device/torch-version guards and skips.
> - Minor: update reference in `vllm/env_override.py` comment to `tests/compile/test_fusion_attn.py::test_attn_quant`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d500caea6c477d94a04eb6e86d76b86da2ad5a94. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Reduces CI flakiness/runtime and streamlines fusion testing.
> 
> - CI: add optional `Hopper Fusion E2E Tests (H100)` (single- and 2xH100) and `Sequence Parallel Tests (H100)`; trim `Distributed Tests (H200)`; tweak filters/commands in `.buildkite/test-pipeline.yaml` and export `VLLM_TEST_CLEAN_GPU_MEMORY` where needed
> - Testing: introduce `tests/compile/fusion_test_utils.py` with shared models/helpers; simplify `tests/compile/distributed/test_fusions_e2e.py` to TP2-focused checks (attn+quant, allreduce+RMSNorm, async TP) using the utilities; add E2E attn-quant in `tests/compile/test_fusion_attn.py` and RMSNorm+group FP8 quant E2E in `tests/compile/test_fusion.py`; add device/torch-version guards and skips
> - Minor: update comment reference in `vllm/env_override.py` to `tests/compile/test_fusion_attn.py::test_attn_quant`
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 1a2de2e3b124633c6f1e90209b8348252ec5b106. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

LucasWilkinson committed:
breakup h200 tests

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

khluu commented on .buildkite/test-pipeline.yaml:
```diff
@@ -1315,6 +1338,19 @@ steps:
+  num_gpus: 2
+  commands:
+    - export VLLM_TEST_CLEAN_GPU_MEMORY=1
+    - nvidia-smi
```
`nvidia-smi` is automatically added so I don't think you need it here

LucasWilkinson committed:
review comments

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
review comments

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson commented on .buildkite/test-pipeline.yaml:
```diff
@@ -1030,6 +1030,29 @@ steps:
     # Run all e2e fusion tests
     - pytest -v -s tests/compile/distributed/test_fusions_e2e.py
 
+- label: Hopper Fusion E2E Tests # 1h+ min
```
@angelayi pointed out that `test_attn_quant` dont actually use more the 1GPU so split those off

LucasWilkinson committed:
format

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

khluu commented on .buildkite/test-pipeline.yaml:
```diff
@@ -1030,6 +1030,47 @@ steps:
     # Run all e2e fusion tests
     - pytest -v -s tests/compile/distributed/test_fusions_e2e.py
 
+- label: Hopper Fusion E2E Tests (H100) # ?min
```
maybe don't leave `?` for now, we can update it after we get more data

khluu reviewed (approved):
Thank you so much! The changes in `test-pipeline.yaml` LGTM! Can you rebase against main?

LucasWilkinson committed:
format

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix: make PR a pure reorg - add missing imports and restore original test behavior

- Add has_cuda_graph_wrapper_metadata to fusion_test_utils.py (was missing)
- Import has_cuda_graph_wrapper_metadata in test_fusion_attn.py
- Add missing 'import vllm.config' to test_fusion.py
- Restore test_rms_group_quant to match original:
  - Add back @pytest.mark.skipif(is_blackwell(), ...)
  - Use model_kwargs['attention_config'] instead of monkeypatch.setenv
  - Use PassConfig(fuse_norm_quant=True, fuse_act_quant=True, ...)
- Remove accidental Llama-4-Scout skip from test_attn_quant

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix: remove trailing newline

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix pre-commit

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix tests

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
fix

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
move

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
clean up

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>

LucasWilkinson committed:
clean

Signed-off-by: Lucas Wilkinson <lwilkins@redhat.com>
